web/src/main/webapp/ui/menu.ts

```diff
@@ -318,7 +378,20 @@ export class ContextMenu extends BaseMenu<MenuItem> implements IHtmlElement {
             y = max.height - this.outer.offsetHeight - 5;
         if (y < 0)
             y = 0;
-        this.move(x, y);
+
+        this.subMenuOffsetWidth = this.longestSubMenuText * this.pixelPerChar;
```

Comment:
what you should do is really measure the width of the object as rendered on the screen, do not attempt to compute the size in this simple way.

Reply:
We can't get the width because the submenu (the cell) is not yet displayed. Now, I approximate the width based on the main menu since the main menu will be displayed and has `offsetWidth` value.